app/controllers/utm_links_controller.rb

```diff
@@ -1,13 +1,109 @@
 # frozen_string_literal: true
 
 class UtmLinksController < Sellers::BaseController
-  before_action :set_body_id_as_app
+  before_action :set_utm_link, only: [:edit, :update, :destroy]
+  before_action :authorize_utm_link, only: [:edit, :update, :destroy]
+  layout "inertia", only: [:index, :new, :edit]
 
   def index
     authorize UtmLink
+
+    route_params = index_route_params_helper
+
+    render inertia: "Analytics/UtmLinks/Index",
+           props: {
+             utm_links_props: PaginatedUtmLinksPresenter.new(
+               seller: current_seller,
+               query: route_params[:query],
+               page: route_params[:page],
+               sort: { key: route_params[:key], direction: route_params[:direction] }.compact
+             ).props,
+             utm_links_stats: InertiaRails.merge { fetch_utm_links_stats(route_params[:ids]) },
+           }
+  end
+
+  def new
+    authorize UtmLink
+
+    utm_link_presenter = UtmLinkPresenter.new(seller: current_seller)
+
+    render inertia: "Analytics/UtmLinks/New", props: {
+      **utm_link_presenter.new_page_react_props(copy_from: params[:copy_from]),
+      additional_metadata: InertiaRails.optional { utm_link_presenter.new_additional_metadata_props },
+    }
+  end
+
+  def edit
+    render inertia: "Analytics/UtmLinks/Edit", props: UtmLinkPresenter.new(seller: current_seller, utm_link: @utm_link).edit_page_react_props
+  end
+
+  def create
+    authorize UtmLink
+
+    save_utm_link
+  end
+
+  def update
+    return head :not_found if @utm_link.deleted?
+
+    save_utm_link
+  end
+
+  def destroy
+    @utm_link.mark_deleted!
+
+    redirect_to utm_links_dashboard_path(index_route_params_helper.except(:ids).compact), notice: "Link deleted!", status: :see_other
   end
 
   private
+    def index_route_params_helper
+      permitted = params.permit(:query, :page, :key, :direction, ids: [])
+
+      {
+        direction: permitted[:direction],
+        ids: permitted[:ids],
+        key: permitted[:key],
+        page: permitted[:page],
+        query: permitted[:query],
+      }.compact
+    end
```

Comment:
Without this table will lose it's current state, this helps preserve navigation and reloads data accordingly